web/src/pages/Courts/CourtDetails/TopSearch.tsx

```diff
@@ -137,18 +174,17 @@ const TopSearch: React.FC = () => {
               type="text"
               placeholder="Search"
               value={search}
-              onChange={(e) => setSearch(e.target.value)}
+              onChange={handleSearchChange}
+              onKeyDown={handleKeyDown}
             />
             {search && filteredCourts.length > 0 && (
               <SearchResultsContainer>
-                {filteredCourts.map((court) => (
+                {filteredCourts.map((court, index) => (
                   <StyledCard
                     key={court.id}
-                    selected={court.id === currentCourtId}
```

Comment:
Thanks for your contribution @ManiBAJPAI22 !

I would like to keep the original logic to highlight the selected court, and make it so such that the `keyboardSelected` style doesn't clash with it (i.e. no primary-blue left-border). Maybe some kind of outline? also, we have a new version of the ui-components library that will eventually substitute all these components, and this new library has functionality already built-in, although we don't have a date for the migration yet.